### PR TITLE
fix #5483: show geotagged images on map

### LINKED ISSUE #5483: Show geotagged log or listing images on the map
Garmin devices have a nice feature for images that provide GPS coordinates via Exif: They place a picture icon on the map that shows a thumbnail of the image when activated. I found this feature very helpful and would therefore propose to add a similar feature to c:geo.

### PR DESCRIPTION
**Pull request by pstorch**
Added an Overlay to the images in the Cachlisting "images" tab if coordinates are available in the image.
Added onClick() and onLongClick() handler to start Navigation.

Unfortunately the Images are parsed twice (first to create the Bitmap and later to read the Exif data). I don't know if this is a problem. I couldn't recognize longer loading times on my device. 
Please test on yours.
If it's a problem maybe we should add a preference to turn it on and off.

### CODE REVIEW

File: main/src/cgeo/geocaching/ui/ImagesList.java [hunk 1]
```diff
@@ -153,17 +170,65 @@ public void onClick(final View arg0) {
...
+                return;
+            }
+
+            final Metadata metadata = ImageMetadataReader.readMetadata(file);
+            final Collection<GpsDirectory> gpsDirectories = metadata.getDirectoriesOfType(GpsDirectory.class);
+            if (gpsDirectories == null) {
+                return;
+            }
+
+            for (GpsDirectory gpsDirectory : gpsDirectories) {
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/ui/ImagesList.java]: Missing `final`

**Comment by samueltardieu** [comment]
I kinda like it.

File: main/src/cgeo/geocaching/ui/ImagesList.java [hunk 2]
```diff
@@ -153,17 +170,65 @@ public void onClick(final View arg0) {
...
             images.put(imageView.getId(), img);
 
             view.invalidate();
         }
     }
 
+    private void addGeoOverlay(final RelativeLayout imageViewLayout, final Image img) {
+        try {
+            // Extract metadata.
+            final File file = LocalStorage.getStorageFile(geocode, img.getUrl(), true, true);
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/ui/ImagesList.java]: Last parameter should be `false`. Also, `getStorageFile` cannot return `null`, I'll add an annotation there.

File: main/src/cgeo/geocaching/ui/ImagesList.java [hunk 3]
```diff
@@ -153,17 +170,65 @@ public void onClick(final View arg0) {
...
+                                    NavigationAppFactory.startDefaultNavigationApplication(2, activity, gpt);
+                                    return true;
+                                }
+                            });
+                        }
+                    });
+                    break; // add only the first found geoLocation
+                }
+            }
+        } catch(Exception e) {
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/ui/ImagesList.java]: Missing space after `catch` and `final` before `Exception`.

**Comment by pstorch** [comment]
I'll amend it this evening.

**Comment by pstorch** [comment]
Fixed the review comments.

**Comment by samueltardieu** [comment]
The failure is not due to your changes. I'll merge.

**Pull request merged**